plugins/modules/idrac_license.py

```diff
@@ -0,0 +1,577 @@
+        "MessageId": "LIC900",
+        "Name": "Export: License",
+        "PercentComplete": 100,
+        "StartTime": "2024-01-09T05:16:19",
+        "TargetSettingsURI": null
+    }
+error_info:
+  description: Details of the HTTP Error.
+  returned: on HTTP error
+  type: dict
+  sample: {
+    "error": {
+      "code": "Base.1.8.GeneralError",
+      "message": "A general error has occurred. See ExtendedInfo for more information.",
+      "@Message.ExtendedInfo": [
+        {
+          "MessageId": "Base.1.8.AccessDenied",
+          "Message": "The authentication credentials included with this request are missing or invalid.",
+          "MessageArgs": [],
+          "RelatedProperties": [],
+          "Severity": "Critical",
+          "Resolution": "Attempt to ensure that the URI is correct and that the service has the appropriate credentials."
+        }
+      ]
+    }
+  }
+'''
+
+import json
+import os
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.compat.version import LooseVersion
+from ansible_collections.dellemc.openmanage.plugins.module_utils.idrac_redfish import iDRACRedfishAPI, idrac_auth_params
+from ansible.module_utils.six.moves.urllib.error import URLError, HTTPError
+from ansible.module_utils.urls import ConnectionError, SSLValidationError
+from ansible_collections.dellemc.openmanage.plugins.module_utils.utils import (
+    get_idrac_firmware_version, validate_and_get_first_resource_id_uri,
+    remove_key)
+
+SYSTEMS_URI = "/redfish/v1/Managers"
+IDRAC_JOB_URI = "{res_uri}/Jobs/{job_id}"
+EXPORT_LICENCE_NETWORK_SHARE_URI = "Oem/Dell/DellLicenseManagementService/Actions/DellLicenseManagementService.ExportLicenseToNetworkShare"
+
+SUCCESS_EXPORT_MSG = "Successfully exported the license."
+SUCCESS_DELETE_MSG = "Successfully deleted the license."
+SUCCESS_IMPORT_MSG = "Successfully imported the license."
+FAILURE_MSG = "Unable to {operation} the licens with id {license_id}."
+
+IGNORE_CERTIFICATE_WARNING = {"off": "Off", "on": "On"}
+PROXY_SUPPORT = {"off": "Off", "default_proxy": "DefaultProxy", "parameters_proxy": "ParametersProxy"}
+PROXY_TYPE = {"http": "HTTP", "socks": "SOCKS"}
+
+
+class DeleteLicense():
+    def __init__(self, idrac, module):
+        self.idrac = idrac
+        self.module = module
+
+    def execute(self, module):
+        license_id = module.params.get('license_id')
+        delete_license_url = f"/redfish/v1/LicenseService/Licenses/{license_id}"
+        delete_license_status = self.idrac.invoke_request(delete_license_url, 'DELETE')
+        status = delete_license_status.status_code
+        share_type = module.params.get('share_parameters').get('share_type')
+        if status == 204:
+            module.exit_json(msg=SUCCESS_DELETE_MSG, changed=True)
+        else:
+            module.fail_json(FAILURE_MSG.format(operation=share_type, license_id=license_id), changed=False)
+        return delete_license_status
+
+
+class ExportLicense():
```

Comment:
Remove small bracket in end of class name, not needed in latest python, check for other class too.

Reply:
Done